dallinger/experiment.py

```diff
@@ -602,27 +608,46 @@ def end_experiment(self):
         return True
 
     def events_for_replay(self, session=None, target=None):
-        """Be default we return all infos in order for replay"""
+        """Returns an ordered list of "events" for replaying.
+        Experiments may override this method to provide custom
+        replay logic. The "events" returned by this method will be passed
+        to :meth:`~Experiment.replay_event`. The default implementation
+        simply returns all :class:`~dallinger.models.Info` objects in the
+        order they were created.
+        """
         if session is None:
             session = self.session
         return session.query(Info).order_by(Info.creation_time)
 
     def replay_event(self, event):
+        """Stub method to replay an event returned by
+        :meth:`~Experiment.events_for_replay`.
+        Experiments must override this method to provide replay support.
+        """
         pass
 
     def replay_start(self):
+        """Stub method for starting an experiment replay.
+        Experiments must override this method to provide replay support.
+        """
         pass
 
     def replay_finish(self):
+        """Stub method for ending an experiment replay.
+        Experiments must override this method to provide replay support.
+        """
         pass
 
     def replay_started(self):
+        """Returns `True` if an experiment replay has started."""
         return True
 
     def is_complete(self):
         """Method for custom determination of experiment completion.
-        Return None to use the experiment server default, otherwise
-        `True` or `False`"""
+        Experiments should override this to provide custom experiment
```

Comment:
Is there a reason to not just return `False` here so the default implementation provides the default behavior directly? (I know the PR is about documentation, not behavior, but the way we do this now makes the documentation puzzling.)

Reply:
The default behavior isn't to return `False` it's to let the experiment server look at the status of working participants and whether networks are filled. If this method returns `False` that will override the logic in the experiment server code.

Reply:
Hmm... This method is called in the `/summary` route, where `/summary` checks to see if the return value is `None`, and if it is, it simply swaps in `False` in its own return value. The behavior you describe --use the return value of `is_complete()` if it's not None, but run some other check if it is-- makes more sense, but that's not what I see happening. Am I missing something? Willing to let this go, since it's only related to the actual documentation story.

Reply:
Aha... yes, I'm missing something:
```
if state['unfilled_networks'] == 0:
        if working == 0 and state['completed'] is None:
```
Nevermind. :)

Reply:
You had me worried for a moment!